project/Http4sPlugin.scala

```diff
@@ -53,7 +87,117 @@ object Http4sPlugin extends AutoPlugin {
     mimaFailOnProblem := http4sMimaVersion.value.isDefined,
     mimaPreviousArtifacts := (http4sMimaVersion.value map {
       organization.value % s"${moduleName.value}_${scalaBinaryVersion.value}" % _
-    }).toSet
+    }).toSet,
+
+
+    http4sBuildData := {
+      val dest = target.value / "hugo-data" / "build.toml"
+      val (major, minor) = http4sApiVersion.value
+
+      val releases = latestPerMinorVersion(baseDirectory.value)
+        .map { case ((major, minor), v) => s""""$major.$minor" = "${v.string}""""}
+        .mkString("\n")
+
+      // Would be more elegant if `[versions.http4s]` was nested, but then
+      // the index lookups in `shortcodes/version.html` get complicated.
+      val buildData: String =
+        s"""
+           |[versions]
+           |"http4s.api" = "$major.$minor"
+           |"http4s.current" = "${version.value}"
+           |"http4s.doc" = "${docExampleVersion(version.value)}"
+           |scalaz = "${scalazVersion.value}"
+           |circe = "${circeJawn.revision}"
+           |cryptobits = "${cryptobits.revision}"
+           |"argonaut-shapeless_6.2" = "1.2.0-M5"
+           |
+           |[releases]
+           |${releases}
+         """.stripMargin
+
+      IO.write(dest, buildData)
+    }
+  ) ++ releaseSettings
+
+  val releaseSettings = Seq(
+    // Reset a couple sbt-release defaults that rig changed
+    releaseVersion := { ver => Version(ver).map(_.withoutQualifier.string).getOrElse(versionFormatError) },
+    releaseTagName := s"v${if (releaseUseGlobalVersion.value) (version in ThisBuild).value else version.value}",
+
+    releaseProcess := Seq(
+      if (!isSnapshot.value)
```

Comment:
Can these be broken out into individual functions for greater clarity? Functional programmer in me looks at this as practically imperative.

Reply:
I tried earlier but struggled a bit with names.  I'll give it another go.

Reply:
I'm not finding it much clearer when I give them names.

Reply:
See if you like 0fc93f6 better.